internal/services/orbital/client/client.go

```diff
@@ -18,19 +20,28 @@ type Client struct {
 	GroundStationClient  *groundstation.GroundStationClient
 }
 
-func NewClient(o *common.ClientOptions) *Client {
-	spacecraftClient := spacecraft.NewSpacecraftClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&spacecraftClient.Client, o.ResourceManagerAuthorizer)
+func NewClient(o *common.ClientOptions) (*Client, error) {
+	spacecraftClient, err := spacecraft.NewSpacecraftClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build orbital spacecraft Client: %+v", err)
+	}
+	o.Configure(spacecraftClient.Client, o.Authorizers.ResourceManager)
 
-	contactProfileClient := contactprofile.NewContactProfileClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&contactProfileClient.Client, o.ResourceManagerAuthorizer)
+	contactProfileClient, err := contactprofile.NewContactProfileClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build orbital contact profile Client: %+v", err)
+	}
+	o.Configure(contactProfileClient.Client, o.Authorizers.ResourceManager)
 
-	contactClient := contact.NewContactClientWithBaseURI(o.ResourceManagerEndpoint)
-	o.ConfigureClient(&contactClient.Client, o.ResourceManagerAuthorizer)
+	contactClient, err := contact.NewContactClientWithBaseURI(o.Environment.ResourceManager)
+	if err != nil {
+		return nil, fmt.Errorf("build orbital contact Client: %+v", err)
```

Comment:
Please fix the error messages here with consistent casing and grammar `building ...`